File: python/find-the-difference.py
class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        char_count = defaultdict(lambda: 0)
        for c in s:
            char_count[c] += 1
        for c in t:
            if not char_count.get(c, 0):
                return c
            char_count[c] -= 1
        return "0"

File: rust/find-the-difference.rs
use std::collections::HashMap;
use std::iter::FromIterator;

impl Solution {

    pub fn find_the_difference(s: String, t: String) -> char {
        // Initialize HashMap where character count will be held
        let mut char_count:HashMap<char,u32> = HashMap::new();

        // Iterate over s and increase the count by one for each char found
        for c in s.chars() {
            match char_count.get(&c) {
                Some(count) => char_count.insert(c, count + 1),
                None => char_count.insert(c, 1)
            };
        }

        // Iterate over t and find the missing element, or decrease the count
        for c in t.chars() {
            match char_count.get(&c) {
                Some(0) => return c,
                None => return c,
                Some(count) => char_count.insert(c, count - 1)
            };
        }

        // Default scenario with zero
        '0'
    }
}
